### DISCUSSION TITLE
How to access SoapClient properties

### DISCUSSION THREAD

**Discussion by GoldXD** [phpro/soap-client#509] [Q&A] [answered]
SoapClient has _getLastRequestHeaders(), __getLastRequest(), __getLastResponseHeaders() and __getLastResponse()
I need to use these for some detailed logging.

I see it's in the Caller, I just don't know how to properly access it.
![image](https://github.com/phpro/soap-client/assets/8411655/039d5881-3fd7-4897-8c83-78059c7bd955)

**Comment by veewee** [comment, marked as answer]
Hello,

You can wrap the transport with a `TraceableTransport` and access that information from that service:
https://github.com/php-soap/ext-soap-engine?tab=readme-ov-file#example-usage

Do note that it's probably a better idea to add an HTTP logger instead: When logging requests and responses at transport level, you don't get to see any manipulations that are being done by other HTTP-level plugins. This gives you a central trustworthy place for you to detect the message contents.

More info:
https://github.com/php-soap/psr18-transport?tab=readme-ov-file#httplug-middleware
